### PR TITLE
Experiment with breaking CI tests up into smaller chunks

### PR DESCRIPTION
**Pull request by jasongrout**
## Code changes

- [x] Temporarily turn off code coverage in CI because it produces lots of errors
- [x] Break up test runs to different CI tasks

### CODE REVIEW

**Comment by telamonian** [comment]
>     * Temporarily turn off code coverage in CI because it produces lots of errors

👍 (and pinging @blink1073)
Me and Steve had a conversation a little while ago in #7940 about the long-running brokenness of coverage. I support turning it off for now.

Overall, I like the approach you're taking in this PR better than the one I took in #7940. If you're splitting up the test suites, you should most definitely split out test-services, since that one is usually responsible for the bulk of failures

**Comment by jasongrout** [comment]
I think this is ready for review. Unfortunately we still see async errors in the services tests, and linkcheck errors in the doc tests, but this makes it easier to diagnose what things are failing and provides better concurrency too.

**Comment by telamonian** [comment]
Hah, a failure in test-services is indeed one of the things currently keeping the CI from passing. Hopefully splitting it out like this will make tracking down the real underlying issue a bit more straightforward.

File: scripts/ci_script.sh [hunk 1]
```diff
@@ -17,11 +17,27 @@ if [[ $GROUP == python ]]; then
 fi
 
 
-if [[ $GROUP == js ]]; then
+if [[ $GROUP == js* ]]; then
+
+    jlpm build:packages
```
Comment (telamonian) [inline scripts/ci_script.sh]: Is this here because the windows JS tests are still run as a single job?

Comment (jasongrout) [reply scripts/ci_script.sh]: Let me see if we can just build a subgroup of the packages. Not sure how well that works..

Comment (jasongrout) [reply scripts/ci_script.sh]: Thanks! That cut a lot of the test runs from 5m to 3m - a huge savings!

**Comment by jasongrout** [comment]
> But why not also split up the Windows jobs?

We're already at 49 different jobs. Might as well go all the way to 100, I guess? Why do we run the js tests on windows anyway? Perhaps we just avoid it, and run the python and installation and system tests on Windows?

When has Windows legitimately found a problem in the js side that the linux tests haven't?

**Comment by jasongrout** [comment]
> But why not also split up the Windows jobs?

Can we take this up on a different PR, and move more gradually step by step? It may be best to get rid of the windows js tests.

**Comment by telamonian** [comment]
> When has Windows legitimately found a problem in the js side that the linux tests haven't?

That meshes with my own experience; doubling up the JS testing on both linux and windows mostly seems to just double the noise. I guess the central question is: do the VMs/browsers we run on top of diverge significantly enough on different platforms that we have to worry about it? I personally haven't run into a case of this for the stuff that the JS tests cover (though of course there are plenty of divergences in how graphical elements get rendered).

I do remember that at some point we went from not having many windows tests to near linux/windows CI parity. Was the motivating factor a nasty bug that could have been caught by comprehensive windows CI, or was it more that adding the windows tests seemed like a good idea at the time.

**Comment by jasongrout** [comment]
> I do remember that at some point we went from not having many windows tests to near linux/windows CI parity. Was the motivating factor a nasty bug that could have been caught by comprehensive windows CI, or was it more that adding the windows tests seemed like a good idea at the time?

I don't remember what the original reasoning was. I think this is a good place to wrap up this ticket, though, and take up the windows js tests in another issue/PR.

**Comment by blink1073** [comment]
Thanks!  Docs failures are covered in #8068.

**Comment by echarles** [comment]
Even with those smaller chunks, I still see GitHub failure on test-services where it should not.

**Comment by jasongrout** [comment]
> Even with those smaller chunks, I still see GitHub Actions failure on test-services where it should not.

Yes, this doesn't seem to solve the problem, but does (a) run the tests more concurrently, and (b) helps pinpoint where the errors occur in a more fine-grained way.

**Comment by echarles** [comment]
Sure, this is a good move to nail down the issue and have a better view on the build :+1:

**Pull request merged**